File: server/spring-boot-ecommerce/src/main/java/com/ecommerce/springbootecommerce/domain/OrderItem.java
package com.ecommerce.springbootecommerce.domain;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.math.BigDecimal;

@Getter
@Setter
@Entity
@Table(name = "order_item")
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    private int quantity;

    private BigDecimal amount;

    @OneToOne
    private Product product;

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;

}

File: client/ecommerce-app/src/app/modules/models/orderItem.ts
import { Product } from './product';

export class OrderItem {
  quantity!: number;
  amount!: number;
  product!: Product;

  constructor(quantity: number, amount: number, product: Product) {
    this.quantity = quantity;
    this.amount = amount;
    this.product = product;
  }
}
